javascript/packages/orchestrator/src/chainSpec.ts

```diff
@@ -663,64 +683,130 @@ export async function getChainIdFromSpec(specPath: string): Promise<string> {
+    [
+      decorators.green("🧪 Mutating chain spec"),
+      decorators.white(substitutedCommandArgs.join(" ")),
+    ],
+  ]);
+
   try {
-    // Relay-chain spec customization logic
-    const plainRelayChainSpec = readAndParseChainSpec(specPath);
-    const keyType = specHaveSessionsKeys(plainRelayChainSpec)
-      ? "session"
-      : "aura";
-
-    // Clear all defaults
-    clearAuthorities(specPath);
-
-    // add balances for nodes
-    await addBalances(specPath, networkSpec.relaychain.nodes);
-
-    // add authorities for nodes
-    const validatorKeys = [];
-    for (const node of networkSpec.relaychain.nodes) {
-      if (node.validator) {
-        validatorKeys.push(node.accounts.sr_stash.address);
-
-        if (keyType === "session") {
-          const key = getNodeKey(node);
-          await addAuthority(specPath, node, key);
+    await new Promise<void>(function (resolve, reject) {
+      if (processes["mutator"]) {
+        processes["mutator"].kill();
+      }
+
+      // spawn the chain spec mutator thread with the command and arguments
+      processes["mutator"] = spawn(
+        substitutedCommandArgs[0],
+        substitutedCommandArgs.slice(1),
+        { cwd: workingDirectory },
+      );
+      // flush the modified spec to a different file and then copy it back into the original path
+      const spec = fs.createWriteStream(chainSpecModifiedPath);
+
+      // `pipe` since it deals with flushing and we need to guarantee that the data is flushed
+      // before we resolve the promise.
+      processes["mutator"].stdout.pipe(spec);
+
+      processes["mutator"].stderr.pipe(process.stderr);
+
+      processes["mutator"].on("close", (code) => {
+        if (code === 0) {
+          resolve();
         } else {
-          await addAuraAuthority(specPath, node.name, node.accounts!);
-          await addGrandpaAuthority(specPath, node.name, node.accounts!);
+          reject(new Error(`Process returned error code ${code}!`));
         }
+      });
 
-        await addStaking(specPath, node);
-      }
-    }
+      processes["mutator"].on("error", (err) => {
+        reject(err);
+      });
+    });
 
-    if (networkSpec.relaychain.randomNominatorsCount) {
-      await generateNominators(
-        specPath,
-        networkSpec.relaychain.randomNominatorsCount,
-        networkSpec.relaychain.maxNominations,
-        validatorKeys,
-      );
-    }
+    // copy the modified file back into the original path after the mutation has completed
+    fs.copyFileSync(chainSpecModifiedPath, chainSpecFullPath);
+  } catch (e: any) {
+    console.error(`\n${decorators.red("Failed to mutate chain spec!")}`);
+    throw e;
+  }
+}
 
-    if (networkSpec.relaychain.genesis) {
-      await changeGenesisConfig(specPath, networkSpec.relaychain.genesis);
-    }
+export async function customizePlainRelayChain(
+  specPath: string,
+  networkSpec: ComputedNetwork,
+): Promise<void> {
+  // Relay-chain spec customization logic
```

Comment:
Regarding the last commit, I believe we shouldn't let the process go on if a step in chain spec customization has failed. By removing the try-catch structure from `customizePlainRelayChain`, the error would go into the `spawn` try-catch structure where it would print it out, dump the logs and exit the process. This was the way it was done prior to displacement of the relay's plain chain spec customization to chainSpec.ts, too.